### PR TITLE
docs: add step-by-step contribution guide

### PR DESCRIPTION
**Pull request by taqui-786**
## Changes

Added Contribuition step-by-step Guide for beginners in contribuiting.md file.

## Context

I am adding a step-by-step contribution guide to contribuition.md. This guide will streamline onboarding for new contributors, enhance collaboration, and ensure consistent contributions. It will serve as a valuable resource, fostering a more inclusive and productive open-source community.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by secustor — CHANGES_REQUESTED** [review]

**Review by taqui-786 — COMMENTED** [review]
Reviewed changes .

File: .github/contributing.md [hunk 1]
```diff
@@ -1,4 +1,66 @@
...
+   ```bash
+   git push origin feature/my-feature
+
+5. **Open a Pull Request:** Open a pull request on the main repository with a clear title and description of your changes.
+
+## Review Process
+
+The maintainers will review your pull request and may request changes or provide feedback.
+
+Once your changes are approved, they will be merged into the main branch.
```
Comment (HonkingGoose) [inline .github/contributing.md:63]:
```suggestion
```

We already have information about reviewing incoming PRs, and behavior rules: https://github.com/renovatebot/renovate/blob/6aaf8df9f2aab12ad1124ca5dd52f3dac1a65f07/.github/contributing.md?plain=1#L24-L74

File: .github/contributing.md [hunk 2]
```diff
@@ -1,4 +1,66 @@
-# Contributing
+# Contributing to Renovate
+
+Thank you for considering contributing to our Renovate App. Your help is greatly appreciated! This guide will help you get started and ensure that your contributions are effective.
```
Comment (HonkingGoose) [inline .github/contributing.md:3]:
```suggestion
Thank you for considering contributing to Renovate.
We appreciate your help!
Read this guide to get started, and to make sure your contributions are effective.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: .github/contributing.md [hunk 3]
```diff
@@ -1,4 +1,25 @@
...
+### Setting up the development environment
+
+Read the [Fork and Clone section](https://github.com/renovatebot/renovate/blob/main/docs/development/local-development.md#fork-and-clone) of the local development docs.
+
+## Submitting Changes
+
+If you've never worked with Git before, please read chapter one, two and three of the [ProGit 2 book](https://git-scm.com/book/en/v2) first.
+
+Then read [ProGit 2, chapter 6.2 GitHub - Contributing to a Project].
+
```
Comment (rarkins) [inline .github/contributing.md:22]: We host this project on _Git_ Hub, I don't think we need to go this far down the stack. Else you could also argue we need to start explaining the english language, use of computers, etc.

Comment (HonkingGoose) [reply .github/contributing.md:22]:
```suggestion
```

Let's get rid of this then. 😄

**Review by secustor — COMMENTED** [review]
Nothing which would lead me to block this. 

Requesting a review from the docs maintainer.

**Comment by rarkins** [comment]
I don't see the point of this PR anymore. It includes two links to the local development doc, but we already have a link to that doc further down the page.

**Review by astellingwerf — COMMENTED** [review]
The new introduction has a rather narrow definition of contribution, heavily focussed on code development. Contributions are also things like code reviews, responding to questions from the community, submitting bug reports and feature requests. These activities don't necessarily have the same prerequisites.

**Pull request closed without merging**